### ISSUE TITLE
Map UTM coordinates

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/arctos#2309]
See https://www.geoplaner.com/

**Comment by mkoo** [comment]
Also bulkloading UTMs
https://github.com/ArctosDB/PG-migration-testing/issues/29

**Comment by dustymc** [comment]
Fully handling UTM should be possible under PG, just need to install postgis (and maybe standardize datum vocabulary and such).

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
What is the status on this? We currently have all our UTMs in locality remarks, which is not great at all since that has been our collection's primary coordinate type. I'm also trying to add our new localities, I've converted the coordinates to lat/long, but I still need to record original UTM. I can put in collecting event, but it does not have a place to record datum.

**Comment by dustymc** [comment]
I recommend a new locality attribute.

I also have UTM "verbatim" in collecting event, and the lack of a locality.verbatim has caused us to stretch that in indefensible directions, but it's really designed for "came in as this, Arctos converted" rather than "the collector scrawled this on the label."

**Comment by dustymc** [comment]
You might also consider some structured data type - JSON or XML or whatever you're comfortable with - if you foresee a reason to need these as more than strings.

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
Yes, we'll definitely need these as more than strings. We need to have fields for zone, northing, easting, and datum. I don't have a preference about what structured data type is used since I know nothing about the differences between those options.

It would be nice to have UTMs automatically convert to lat/long as Orig_Lat_Long_Units does in collecting event. Not sure that model would even work for us though since, because of the nature of our localities, we enter locality first.

Basically, having the UTMs either converted automatically/mappable would save us the task of having to convert roughly 8000 UTMs in some external program and then reloading those back into Arctos. Even if we did do that, we still need someplace to keep the original UTMs with datum.

**Comment by dustymc** [comment]
I'm drastically upping priority.

https://arctos.database.museum/info/ctDocumentation.cfm?table=ctlat_long_units should contain nothing that can't be converted to DD.dd format, which is the de facto standard at this point. I can't convert UTM. It's been sorta-almost-maybe worth dealing with the complexities that introduces because we've been abusing "verbatim" (=as-entered). We no longer have reason to do that (https://github.com/ArctosDB/arctos/issues/2930), so it's time to wean ourselves of this. There are two options:

* just drop UTM, which I don't believe will change anything anywhere (other than procedures perhaps)
* https://github.com/ArctosDB/arctos/issues/2660, after which we should be able to convert UTM to DD.dd.

I'm not sure https://github.com/ArctosDB/arctos/issues/2660 is realistic with the current dev environment. I have no idea if TACC can support postgis in production (probably??). I don't think anything can be lost by stopping the not-quite-verbatim-mixed-with-other-stuff crazy now, and adding back UTM when/if we can treat it as spatial, so perhaps this is just a proposal to do whatever we need to do in order to get UTM out of ctlat_long_units without losing information.

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
I am ok with moving UTM from "verbatim" coordinates in collecting event to a locality attribute. However, I need that locality attribute to have a structured data format. Unstructured is useless to me and will just be a mess that we have to reformat later.

UTMs are the primary coordinate format used by our curators and the BLM, so it is imperative for us that this data be in a usable format. If we can eventually get #2660 to work for us, that would be a huge help. Is postgis the only way to get UTM conversion? Google Earth can do it, no problem as long as its in WGS84, so I would think there would be something out there????

Also, I know I could pull this potential new UTM attribute in a query using SQL, but it would be really helpful to be able to pull up this and other locality attributes as separate fields in the locality search.

**Comment by dustymc** [comment]
@Nicole-Ridgwell-NMMNHS you can just create a new locality attribute and enter eg

```
[{"utm_zone":"bla"},{"utm_northing":"12345"},{"utm_easting":"98765"}]
```

Let me know the attribute and your structure and I can set up a validation function.

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
I'll need someone to show me how to create a new type of locality attribute, unless there is a how-to somewhere that I've missed.

I can create some sort of structure for the data and validation is good, but that still doesn't help with querying. I'm still in a position where I have to run spatial queries through our old Access database, which is quickly getting out of date. I shouldn't have to do that.

Look, it would be one thing if UTM were an outdated coordinate system. But UTM is very commonly used, has advantages over Lat/Long when 2D maps are being used, and it is preferred by the federal agencies we work with. Even if coordinate conversion isn't an option, at minimum I need a way to do what I do in Access, which is search for northings and eastings within a range of values.

I'm sorry if I sound frustrated and perhaps I should have been pushing the UTMs issue more, but I've just had to answer a couple of big spacial queries and I don't like that I have to go back to our 2019 data to do that.

**Comment by dustymc** [comment]
>how to create a new type of locality attribute

https://github.com/ArctosDB/arctos/issues/new?assignees=&labels=Function-CodeTables&template=authority-request.md&title=Code+Table+Request+-+

>still doesn't help with querying.

I don't have an immediate solution for that. I can get you a place where your data aren't getting mixed with dissimilar data, and I can type them to meet whatever conditions you want to lay out. Beyond that requires more discussion.

> build SQL query to look for localities within certain ranges 

If you mean locality.dec_lat/dec_long then yes. If you mean with UTM then I think still yes, but it'll probably take me more than a few minutes.

**Comment by dustymc** [comment]
>particular way to format the attribute

Something from which I can extract data - the example I gave can be treated as a data object.
```

arctosprod@arctos>> select jsonb_pretty('[{"utm_zone":"bla"},{"utm_northing":"12345"},{"utm_easting":"98765"}]'::jsonb);
          jsonb_pretty           
---------------------------------
 [                              +
     {                          +
         "utm_zone": "bla"      +
     },                         +
     {                          +
         "utm_northing": "12345"+
     },                         +
     {                          +
         "utm_easting": "98765" +
     }                          +
 ]
(1 row)

Time: 1.120 ms

```

>similar to how we have Township/Range) & rely on the date 

Those are probably less stable, but they're also more immediately accessible. I have no objections to that, but you'll need to be careful in how you enter them. If you do so, then we can convert them to something better at any time and what we do now isn't terribly important.

If you have to be paranoid anyway - and you do! - you could also do that as predictable strings, which I could also access easily - I don't care about the format, as long as it's always the same. If those are predictable enough - eg, 5 numbers then a letter then a space then.... - then I might even be able to control them, which would guarantee some level of functionality.

**Comment by Jegelewicz** [comment]
> as long as it's always the same

That will NEVER happen. (OK maybe not NEVER, but it is highly unlikely that it will happen over time). In my bulkload files, UTM are formatted as "utm: zone 13 3834482N 561347E" and that is included in the locality remark. They SHOULD be consistent because I used a concatenation formula to create the bulkloaded remarks, but there are no guarantees).

> create separate attributes for northing, easting, zone, and datum (similar to how we have Township/Range) & rely on the date fields to sort them out if more than one is entered?

I think that would be best for now.

**Comment by dustymc** [comment]
>That will NEVER happen.

I can make it happen, if ya'll can define a "datatype" that I can check.

**Comment by Jegelewicz** [comment]
> Part of what I like about the separate attributes is that it would be easy to have code tables for zone and datum. And then maybe we could add a data check to make sure easting is a 6 digit integer and northing a 7 digit integer?

Big 10-4 on that.

**Comment by dustymc** [comment]
I think "zone "&##&" "&#######&"mN "&######&"mE "&AAA&" "&## is slightly worse than a JSON object, and slightly better than multiple types theoretically united by date, but who knows. As long as this doesn't turn into you asking for something the data can't support, I'm willing to give it a go.

If I can find a datum value in your "data object" (string or JSON or whatevevr) then I can check it against https://arctos.database.museum/info/ctDocumentation.cfm?table=ctdatum. I could do the same with zone, but that's a code table request.

You can type individual attributes with the UI, I can add additional checks if that's not sufficient.

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
> slightly worse than a JSON object

Sorry, I had meant to use that format and forgot. So :
[{"utm_zone":"##"},{"utm_northing":"#######"},{"utm_easting":"######"},{"utm_datum":"use code table value here"]
with rules:
utm_zone: 2 digit integer
utm_northing: 7 digit integer
utm_easting: 6 digit integer
utm_datum: must match a value in [datum code table](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctdatum)

Visually not the most appealing thing and definitely awkward for data entry, but I like that I wouldn't have separate date, remark, etc. fields for each.

Downside is that if we eventually get a better search display for localities in the UI no one could sort by these or download them in individual columns, it would have to be done with SQL.

**Comment by dustymc** [comment]
>rules:...utm_zone: 2 digit integer...

That should not be much problem to write.

>Visually not the most appealing thing

Your browser understands JSON - manipulating those strings into something else (a table might work for this) is (in theory) technically fairly trivial (but still takes time, of which I seem to have negative reserves lately).

>eventually get a better search display for localities in the UI

In theory, again, there's lots of magic to be had there too, but practically I'm not so sure - I spent a day or two banging my head on that a while back and didn't get very far. I imagine future versions of PG will have more capabilities, but who knows. Even if that never meets reality, having the data in a typed object should ensure that they're extractable if we ever come up with a better solution.

Should I get started on a check function using the format you gave above, or does this need more discussion?

**Comment by Jegelewicz** [comment]
I think this needs more discussion.

Why the heck don't we just create the appropriate fields for UTM?

@ccicero @campmlc @mkoo

**Comment by dustymc** [comment]
>Why the heck don't we just create the appropriate fields for UTM?

Elaborate, please.

**Comment by Jegelewicz** [comment]
We have been beating this dead horse forever - https://github.com/ArctosDB/arctos/issues?q=is%3Aissue+utm

EVERY MSAccess/FileMakePro/Excel "database" has these fields, why are we so resistant to them?

**Comment by Jegelewicz** [comment]
In Collecting Event we should have the following:

utm_zone
utm_northing
utm_easting
utm_datum

**Comment by dustymc** [comment]
That would be a big step away from viewing place data as **place data** and towards viewing them as attributes of catalog records. I'm not sure that's _wrong_ but it's very _different_. It's certainly not something that can be resolved way down here. 

Adding them as collecting event attributes allows that viewpoint without forcing it down anyone else's throat.

**Comment by campmlc** [comment]
So is there a problem with adding the following as separate event
attributes, with controlled vocab as needed? I agree that capturing these
data should not be this hard, and it is clearly a community need. If we
have to tweak the data model to meet global and community needs for data
capture, then so be it.
utm_zone
utm_northing
utm_easting
utm_datum

**Comment by dustymc** [comment]
>there a problem with adding

No, but there's a problem with retrieving from that.

There are no problems with capturing anything in the current model. 

There is a discussion on how things should be captured so they're most usable. That'll probably evolve into UI at some point, but as far as I know the only immediate concerns involve the bulkloader.

**Comment by Jegelewicz** [comment]
We can concatenate up some record attributes into this:

![image](https://user-images.githubusercontent.com/5725767/106014396-b0000500-607a-11eb-8c41-d198c03360ad.png)

It seems that all we need is something to link those four attributes so that they could be presented similarly? Date and Determiner could be that link?

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
Currently the only structured way to enter UTM is in collecting event, which is why I had immediate concerns with the bulkloader.

However, actually having that data in collecting event is useless to me. I don't just need to be able to document UTMs, I need to be able to use them as actual spatial data on a locality level. When they're in collecting event, I have to manage one coordinate in a bunch of different events, which just isn't feasible. Collecting event is supposed to contain the "verbatim" coordinate, but I need one "authoritative" coordinate for each locality. 

A large portion of our locality coordinates are only in UTM. Mass converting these coordinates would solve that, but I don't have the resources for that yet, plus I'd still have to convert them back frequently because most of the people using our collection work with UTM.

With the current state of things (UTM in collecting event), I have to use our old Access database from 2019 to respond to requests for spatial queries, which is not ok or feasible for much longer.

**Comment by dustymc** [comment]
>Date and Determiner could be that link?

That's not a terribly stable link, whether it's stable enough is a curatorial call. FWIW I don't see it as particularly dangerous, especially with the archive logging changes, but I'd also not want to imply in any way that machines might be able to reliably reassemble those data. 

That model also can't guarantee all of the parts exist - there's no realistic way I could prevent you from entering 864 values of "utm_northing" and nothing else.

Given good documentation so users have an understanding of the limitations, I don't object.

**Comment by Jegelewicz** [comment]
> Currently the only structured way to enter UTM is in collecting event, which is why I had immediate concerns with the bulkloader.
> 
> However, actually having that data in collecting event is useless to me. I don't just need to be able to document UTMs, I need to be able to use them as actual spatial data on a locality level. When they're in collecting event, I have to manage one coordinate in a bunch of different events, which just isn't feasible. Collecting event is supposed to contain the "verbatim" coordinate, but I need one "authoritative" coordinate for each locality.
> 
> A large portion of our locality coordinates are only in UTM. Mass converting these coordinates would solve that, but I don't have the resources for that yet, plus I'd still have to convert them back frequently because most of the people using our collection work with UTM.
> 
> With the current state of things (UTM in collecting event), I have to use our old Access database from 2019 to respond to requests for spatial queries, which is not ok or feasible for much longer.

If these were separate COLLECTION event attributes that could be searched in the same way we can search locality attributes:

![image](https://user-images.githubusercontent.com/5725767/106017630-e2f7c800-607d-11eb-9916-8ecc235a4f48.png)

I think that would get closer to what Nicole needs, but I am not sure it will get her everything she needs (searching a range of zones, northings or eastings). It could also mean that she ends up with localities that have events with differing UTM, but I think that just speaks to the need for research on those events.

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
> I am not sure it will get her everything she needs

It gets me pretty much nothing that I need.

> It could also mean that she ends up with localities that have events with differing UTM

Right now I have one UTM per locality, which are stored in multiple collecting events. If I end up with differing UTMs because managing one "authoritative" coordinate in 50 different events is difficult, then I no longer have one "authoritative" UTM, I just have a mess.

**Comment by dustymc** [comment]
>that could be searched in the same way we can search locality attributes:

Not much problem to add that, I've been ignoring it because it's one of our many "emergencies" that never actually got used.

>searching a range 

If you want to do numeric things, you need numeric data. There can't be a "range" operation on ``3834482N`` which can't be anything other than a string. Individual attributes (with units) are again probably the most immediately usable way to get that, although I could (in theory) extract "fields" from a JSON object and cast them to integers.

>ends up with localities that have events with differing UTM,

That's just a "feature" of the model - lots of fairly precise event data gets tossed into relatively imprecise localities for various reasons.

>Right now I have one UTM per locality

That simplifies reassembly as long as it remains true.

>no longer have one "authoritative" UTM, I just have a mess.

So this is a request for a/some LOCALITY attribute(s), not collecting_event, correct?

**Comment by Jegelewicz** [comment]
> managing one "authoritative" coordinate in 50 different events is difficult, then I no longer have one "authoritative" UTM, I just have a mess.

@dustymc Can event attributes be bulk managed? Or SHOULD these be a locality attribute?

> There can't be a "range" operation on 3834482N which can't be anything other than a string. 

If the data is in a field "utm_northing" then the "N" would not be part of the data.

**Comment by dustymc** [comment]
> Can event attributes be bulk managed? 

"Could" anyway - I'm not sure what's available now, we can build tools if there's a real-world use case for them.

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
> So this is a request for a/some LOCALITY attribute(s), not collecting_event, correct?

Yes. I'm ok with creating some sort of collecting event attribute for verbatim UTM, but what I actually need right now is an authoritative locality level UTM.

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
> Individual attributes (with units) are again probably the most immediately usable way to get that, although I could (in theory) extract "fields" from a JSON object and cast them to integers.

If it will be easier to do numeric things with individual attributes, then lets just individual attributes.

**Comment by dustymc** [comment]
>authoritative locality level UTM.

Is this just a conversion problem? Given enough tools and UTM data, would/could/should you

* verbatim_locality="bla bla something UTM={verbatim}"
* convert UTM to DD.dd for storage in table locality
* convert locality DD.dd data to UTM when someone wants that instead

**Comment by Jegelewicz** [comment]
OK, so how about we create the following LOCALITY attributes with rules:
utm_zone: 2 digit integer
utm_northing: 7 digit integer
utm_easting: 6 digit integer
utm_datum: must match a value in datum code table

AND then we would link these up using date and determiner to look like:


UTM |   |   |   |   |  
-- | -- | -- | -- | -- | --
zone | northing | easting | datum | determiner | determination date
13 | 1111111 | 111111 | unknown | jegelewicz | 2021-01-27

What else needs doing to make this work?

**Comment by Jegelewicz** [comment]
> Given enough tools and UTM data, would/could/should you

>> I'd still have to convert them back frequently because most of the people using our collection work with UTM.

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
> OK, so how about we create the following LOCALITY attributes

...

> AND then we would link these up using date and determiner to look like:

Yes

> Given enough tools

Eventually yes, ideally Arctos would auto convert for me, but probably we'll just end up with a GIS geodatabase of our localities, auto-convert everything to lat/long for Arctos & have a process for reconciling the two databases once a month. That GIS functionality is probably at least a couple years away for us right now, and in the meantime, I need to be able to retrieve, in a usable way, the UTMs in Arctos.

**Comment by dustymc** [comment]
>ideally Arctos would auto convert

That's what I was attempting to imply with "enough tools." I was really asking if we can solve this by prioritizing https://github.com/ArctosDB/arctos/issues/2660, which would let us convert to and from anything that postgis understands on demand. (It would also let us accept UTM as "as-entered" - not sure if that's relevant to your needs, but it's in there anyway!)

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
> I was really asking if we can solve this by prioritizing #2660

Yes, that would be the ideal solution to this.

**Comment by dustymc** [comment]
>ideal solution

Then please prioritize that issue! I'll ping TACC to see what we'd need to do to get postgis.

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
In the meantime, if this is the way we're going, can we add UTM back to the specimen event bulkloader?

**Comment by dustymc** [comment]
That still seems most-evil to me, but I'm just pushing buttons...

If we do that it'll be a few days to undo what I've done, then we can pull some ambiguous data, then a few more days to un-un-do the changes, and I don't think any of that helps do what you say you need to do.

Suggest:

* new 'verbatim UTM' or similar attribute of some sort
* prioritize postgis - I think there's a meeting this week
* do cool stuff with the unambiguous data you've stashed in your attribute
* drop the attribute (or don't, whatever)
* yay everybody!

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
Well, at the moment we have ~12,000 localities with UTM data in 1) locality remarks (concatenated with other data, no datum) and 2) Collecting event (structured, but repeated across x number of collecting events). Once we get postgis,working that can be converted (pick one collecting event for each and convert that?).

If the ~300 localities I'm trying to enter now are entered in the same way they can be mass converted with everything else. Or, if I go the temporary attribute route then that will be a separate conversion.

**Comment by dustymc** [comment]
Yes I can help you dig those data out of wherever they end up, and if we've already got to dig part of them out of locality remarks maybe it makes the most sense to put them all there (preferably in some easy-to-extract format).

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
After we add postgis, we're not getting rid of "verbatim" coordinates in collecting event, including UTMS, correct? As long as we're keeping those fields, I still need to enter data into them.

Reading through #2930
collecting_event.verbatim coordinates = what was entered into Arctos

So I should still use those fields to keep track of how the coordinate was originally supplied to Arctos. Which means I need UTMs added back into the specimen bulkloader.

**Comment by dustymc** [comment]
Collecting event verbatim coordinates is "what was fed to Arctos, then converted to DD.dd for locality." Except I can't convert UTM, plus it's editable, so it's really a random jumble of random things. With postgis I will be able to convert, so it can be treated like DD MM SS N/S (or anything else convertible).

Without postgis, there was some crazy thing I did to treat UTM as "verbatim." With postgis it'll just be input, like anything else I can convert. I just don't want to add the 'crazy' back in!

Verbatim exists because I don't have tools and I didn't trust the stuff I wrote to convert very much. With real tools, I'm not sure the concept makes sense at all - we should just be able to convert whatever to whatever whenever we want using tools that have been rigorously tested by thousands/millions of users. I don't see a reason to rush to get rid of it, but I'm not sure it'll be around (at least not serving the same purpose) in a year or two either. 

Some sort of 'verbatim' (beyond/beside verbatim locality) may make some kind of sense in your workflow, but that's easy under an attribute model.

Make sense?

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
Ok, I think I see where you are coming from now.

**Comment by ebraker** [comment]
Chiming in that UCM would LOVE a UTM converter. When we migrated to Arctos we had 29K records with UTMs generated by the MapSTEDI project. Currently those are formatted with a standardized Locality_Remark, e.g., "LatLongRemarks:  UTM Zone: 13, northing 4442980, easting 483435" so that one day we can pull them out and push them through a calculator.

**Comment by ccicero** [comment]
Ditto! We are currently dealing with ~1500 Spotted Owl blood samples, getting data ready to upload. Coords are all in UTMs.

**Comment by dustymc** [comment]
I think this can probably be closed.

* locality and/or collecting event attributes can be used for whatever you consider verbatim
* https://github.com/ArctosDB/arctos/issues/2660 will let us treat UTM in the same way we treat DMS etc.
* I've claimed collecting_event.verbatim_locality for "as-entered"

I'm not sure what's going on with postgis on production, I thought it just needed a scheduled reboot, maybe ping Chris @mkoo .

>UTM Zone: 13, northing 4442980, easting 483435

That is unfortunately, as far as I can determine, insufficient information - see https://github.com/ArctosDB/arctos/issues/3415 and https://arctos.database.museum/info/ctDocumentation.cfm?table=ctutm_zone.

**Comment by Jegelewicz** [comment]
Dropping this here, because this is obviously something a lot of people are interested in.

> Hi Mariel,
> 
> I'm afraid if anyone wants to know about the original coordinates for these records, they will have to look in the collection.
> I strongly encourage the development team to restore the functionality of the database to allow us to continue to use original coordinate formats (and get back to being able to use UTMs again). Luckily converting within degree formats can easily be done in a spreadsheet, not like UTMs that need an online converter. I can't imagine what was so beneficial from this change that justifies losing coordinate entry functionality.
> 
> Thanks for your help in getting this sorted out for me.

Andy

A lot more in the entire email thread
[Fwd_ bulk appending records.pdf](https://github.com/ArctosDB/arctos/files/7650736/Fwd_.bulk.appending.records.pdf)

**Comment by dustymc** [comment]
Data would be most useful.

> omits higher geography,

In next release.

>Arctos only accepting decimal degrees format. 

That is not and has never been true, and the recent issues with DD MM.mm format do not affect this tool.

> verbatim coordinates are in a different format, we have to put the verbatim into the verbatim locality 

Yes, that's always been the recommendation - https://handbook.arctosdb.org/documentation/collecting-event.html#verbatim-locality. 

> made a point of not having coordinates in verbatim.

I've been recommending some locality or event attribute for that situation - just needs an Issue.

>because that bulkload collecting event template had only decimal degrees

An Issue can change that (or lead to documentation or whatever).

>encourage the development team to restore the functionality of the database to allow us to continue to use original coordinate formats (and get back to being able to use UTMs again)

This is and always has been an administrative, not development, issue. I need tools, I don't have the ability to get them for myself or I'd have done so long ago. (Monday, I hope....)

**Comment by Jegelewicz** [comment]
> This is and always has been an administrative, not development, issue. I need tools, I don't have the ability to get them for myself or I'd have done so long ago. (Monday, I hope....)

Yeah - but lots of users don't participate here and so don't know that...just wanted to keep all of the stuff somewhere I could find it if needed.

**Comment by Jegelewicz** [comment]
>     made a point of not having coordinates in verbatim.
> 
> I've been recommending some locality or event attribute for that situation - just needs an Issue.

#2930 was that issue but what we decided there is not working because "as entered" does not accept UTM or degrees decimal minutes or PLSS. We can revive that issue if it is going to be needed. If the long awaited tools will handle these, then perhaps we need to wait for that?

**Comment by campmlc** [comment]
Here is data from the data file that lead to Andy's comment. - guid not included.
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/MCampbell/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/MCampbell/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



verbatim_date | verbatim_locality | coll_event_remarks | began_date | ended_date | locality_name | higher_geog | spec_locality | dec_lat | dec_long | minimum_elevation | maximum_elevation | orig_elev_units | min_depth | max_depth | depth_units | max_error_distance | max_error_units | datum | locality_remarks | georeference_source | georeference_protocol
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
25 May 2018 | New Mexico: Eddy County;   Guadalupe Mountains | 2018-05-25 | 2018-05-25 |   | North America, United States, New Mexico, Eddy County | Guadalupe Mountains | 32.03735 | 104.7825 | 2128 | 2128 | m |   |   |   | 15 | m | World Geodetic System   1984 | GPS (Transcribed) | not recorded



</body>

</html>

**Comment by Jegelewicz** [comment]
I think those headers don't line up with the data?

**Comment by Jegelewicz** [comment]
Also don't see any UTM or other verbatim coordinates?

**Comment by campmlc** [comment]
Try again - here are two records that the specimen event loader kicked out:
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/MCampbell/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/MCampbell/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



spec_locality | collecting_method | collecting_source | verbatim_date | verbatim_locality | began_date | ended_date | habitat | coll_event_remarks | locality_name | datum | max_error_distance | max_error_units | georeference_source | georeference_protocol | orig_lat_long_units | dec_lat | dec_long | lat_deg | long_deg | dec_lat_min | dec_long_min | lat_min | lat_sec | lat_dir | long_min | long_sec | long_dir | orig_elev_units | minimum_elevation | maximum_elevation
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
West side   of Rio Grande 300 m S Montano Rd on Bosque trail system | wild | 2 October 2018 | New Mexico: Bernalillo County; West side of Rio Grande 300 m S Montano Rd   on Bosque trail system | 2018-10-02 | 2018-10-02 | on trail |   |   | World Geodetic System 1984 | 100 | m | GPS (Transcribed) | not recorded | deg. min. sec. |   |   | 35 | 106 |   |   | 8 | 35.3 | N | 40 | 51.2 | W |   |   |  
Guadalupe   Mountains | wild | 25 May 2018 | New Mexico: Eddy County; Guadalupe Mountains | 2018-05-25 | 2018-05-25 | in narrow mixed conifer, maple, oak   canyon | World Geodetic System 1984 | 15 | m | GPS (Transcribed) | not recorded | degrees dec. minutes |   |   | 32 | 104 | 2.241 | 46.95 |   |   | N |   |   | W | m | 2128 | 2128



</body>

</html>

**Comment by campmlc** [comment]
I'll forward the original file to @dusty via email. It's in the email string I forwarded earlier.

**Comment by dustymc** [comment]
>users don't participate here and so don't know

Yes, that's come up a lot lately, I obviously have no idea how to resolve it but users being frustrated because I'm not giving them something I don't know they want isn't ideal....

Maybe we should again try confining the chatter to elsewhere and saving this for simple functional details-worked-out-elsewhere messaging, or set up some other repo for that, or ????????????

> not working because "as entered" does not accept UTM 

I think there's still some significant confusion. "As-entered" is "converted by Arctos" and that's it. If I can't convert it, then it doesn't belong there - it's nothing like "verbatim" and I regret the name. It should probably never have been exposed to users (it's only useful if a script breaks, sorta), the super awkward halfassed UTM-but-not-really workaround that's now polluted the data should never have been allowed, nondigitized datums can't use this concept as soon as I get postgis, etc., etc., etc. Ain't "verbatim" and never has been, despite the bad name. If you need verbatim, then you need a new concept.

"As-entered" will start accepting UTM maybe Monday, but it still wont' be "verbatim." (Something like 95% of the UTM data in Arctos doesn't have sufficient information to be converted, I suspect that's how it comes in; that new concept would be useful for the 'this is what the gave us before we guessed at the zone and datum' data.) 

Thanks @campmlc  - it's not in the email I received (and that's not my github handle).

**Comment by dustymc** [comment]
OK, got data - my first guess from there is always "Excel mangled something." Do you have the CSV with the errors?

**Comment by campmlc** [comment]
I'll ask - and sorry about the handle. That's what happens when I respond to github via gmail - yes, difficult to juggle so many communications platforms. But Github keeps much better track of our issues than gmail ever could.

**Comment by Jegelewicz** [comment]
> I'm not giving them something I don't know they want

Don't take it personally - it is really THE COMMUNITY not giving them something they want. As long as stuff goes on in email threads, only the few people on the thread know what the problem is. I can complain about Excel until my face turns blue, but if I never post to Microsoft customer support - how will they ever know that there is a problem?

**Comment by Jegelewicz** [comment]
> Maybe we should again try confining the chatter to elsewhere

It doesn't matter WHERE we do this IMO - some people will just not participate.

**Comment by campmlc** [comment]
But we can assist the community at large by posting concerns raised by
email here. Not everyone is comfortable with github as a platform -
honestly, it was designed for code, and the user interface takes some
getting used to.
In reference to the issues raised, here we need to post issues regarding
the need for 1) true, verbatim coordinates - not "as entered", with all its
confusing implications - this term should not be publicly visible, 2) the
specimen event bulkloader should not have fields for coordinate formats
that are going to throw an error when the file is loaded. What am I missing?

**Comment by Jegelewicz** [comment]
>  the specimen event bulkloader should not have fields for coordinate formats that are going to throw an error when the file is loaded. What am I missing?

Agree - the issue is the degrees decimal minutes - if we can't use that, then we need to take it out of the template. I've had to explain this to at least three people in the past week. People use this format and have for a long time and suddenly it doesn't work (although suddenly excluding it from the template will probably generate the same issues.)

**Comment by Jegelewicz** [comment]
> But we can assist the community at large by posting concerns raised by email here.

Agree - this is why I post people's email all over Github along with whatever responses I send.

**Comment by dustymc** [comment]
>Github keeps much better track of our issues than gmail

Absolutely, and anything else we've tried. I love github, but I can see how it could quickly become overwhelming too. I don't think "where" matters (although it probably does to someone), but "how." (And no, I don't know how to do better, but I still think I see a problem, maybe related to the sheer volume.)

>this term should not be publicly visible

I'm hoping to file that issue next week or so, I'm glad to see some pre-agreement!

>the specimen event bulkloader should not have fields for coordinate formats
that are going to throw an error when the file is loaded.

That's what I'm confused about: that's really the case, it it/was temporarily the case, it's completely unrelated and not my problem, it's completely unrelated and totally my problem - I'm guessing without data, I don't have the bandwidth to really investigate right now, and there are 608 issues for this to get lost in.

>degrees decimal minutes

Only in a different tool?? https://github.com/ArctosDB/arctos/issues/3916

**Comment by Jegelewicz** [comment]
@dustymc as far as I know you CANNOT load degrees decimal minutes anywhere (although you used to be able to), but it is still an option both in the code table and the templates. Bound to cause problems.

https://arctos.database.museum/info/ctDocumentation.cfm?table=ctlat_long_units#degrees_dec__minutes

![image](https://user-images.githubusercontent.com/5725767/144665242-6c0c9137-b962-43b8-b436-d5dea69595cc.png)

The fact that the ONLY coordinates you can load when bulkloading a locality or event are decimal degrees does not allow for "as entered" in these situations - why are they any different than an entire catalog record?

![image](https://user-images.githubusercontent.com/5725767/144665541-a15bb619-19c7-4ead-a60e-6384a6c346e1.png)

**Comment by dustymc** [comment]
That may or may not be true now, I'm too overwhelmed to look at the moment, I will get what's in test (which handles DM.m) out early next week whether I have postgis (for UTM) or not.

>bulkloading a locality

There's absolutely no place for anything other than DD.d coordinates in locality (and I plan to enhance that to "in WGS84" as soon as I get postgis).

>or event 

No technical reason that can't accept anything I can convert - Issue (if there's not one - or 10....).
